cmd/bucket-targets.go

```diff
@@ -489,31 +489,27 @@ func (sys *BucketTargetSys) UpdateAllTargets(bucket string, tgts *madmin.BucketT
 	defer sys.Unlock()
 
 	// Remove existingtarget and arn association
-	if tgts, ok := sys.targetsMap[bucket]; ok {
-		for _, t := range tgts {
+	if stgts, ok := sys.targetsMap[bucket]; ok {
+		for _, t := range stgts {
 			delete(sys.arnRemotesMap, t.Arn)
 		}
 		delete(sys.targetsMap, bucket)
 	}
 
-	// No need for more if not adding anything
-	if tgts == nil || tgts.Empty() {
-		globalBucketMonitor.DeleteBucket(bucket)
-		return
-	}
```

Comment:
we should delete the arn specific throttle if the target is no longer valid - to avoid leaving around stale entries in the map

Reply:
that is already taken by DeleteBucketThrottle() @poornas in updateBandwidthLimit() in bucket-targets.go

I can't image any where UpdateAllTargets() receives tgts to be nil or its empty.